runtime-modules/content/src/lib.rs

```diff
@@ -1261,6 +944,548 @@ decl_module! {
+            actor: ContentActor<T::CuratorGroupId, T::CuratorId, T::MemberId>,
+            video_id: T::VideoId,
+            metadata: Metadata,
+        ) {
+
+            // Ensure given video exists
+            let video = Self::ensure_video_exists(&video_id)?;
+
+            // Ensure auction for given video id exists, retrieve corresponding one
+            let auction = video.ensure_nft_auction_state::<T>()?;
+
+            // Ensure open type auction
+            auction.ensure_is_open_auction::<T>()?;
+
+            // Ensure there is a bid to accept
+            let last_bid = auction.ensure_last_bid_exists::<T>()?;
+
+            // Ensure actor is authorized to accept open auction bid
+            Self::authorize_nft_owner(origin, &actor, &video)?;
+
+            if let Some(owned_nft) = &video.nft_status {
+
+                let owner_account_id = Self::ensure_owner_account_id(&video, &owned_nft)?;
+
+                let last_bidder_account_id = Self::ensure_member_controller_account_id(last_bid.bidder)?;
+
+                //
+                // == MUTATION SAFE ==
+                //
+
+                let video = Self::complete_auction(video, last_bidder_account_id, owner_account_id);
+
+                // Update the video
+                VideoById::<T>::insert(video_id, video);
+            }
+
+            // Trigger event
+            Self::deposit_event(RawEvent::OpenAuctionBidAccepted(actor, video_id, metadata));
+        }
+
+        /// Issue NFT
+        #[weight = 10_000_000] // TODO: adjust weight
+        pub fn issue_nft(
+            origin,
+            actor: ContentActor<CuratorGroupId<T>, CuratorId<T>, MemberId<T>>,
+            video_id: T::VideoId,
+            royalty: Option<Royalty>,
+            metadata: Metadata,
+            to: Option<T::MemberId>,
+        ) {
+
+            // Ensure given video exists
+            let video = Self::ensure_video_exists(&video_id)?;
+
+            // Ensure have not been issued yet
+            video.ensure_nft_is_not_issued::<T>()?;
+
+            // Ensure channel exists, retrieve channel owner
+            let channel_owner = Self::ensure_channel_exists(&video.in_channel)?.owner;
+
+            ensure_actor_authorized_to_update_channel::<T>(origin, &actor, &channel_owner)?;
+
+            // The content owner will be..
+            let nft_owner = if let Some(to) = to {
+                NFTOwner::Member(to)
+            } else {
+                // if `to` set to None, actor issues to himself
+                Self::actor_to_nft_owner(&actor)?
+            };
+
+            // Enure royalty bounds satisfied, if provided
+            if let Some(royalty) = royalty {
+                Self::ensure_reward_account_is_set(video.in_channel)?;
+                Self::ensure_royalty_bounds_satisfied(royalty)?;
+            }
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            // Issue NFT
+            let mut video = video;
```

Comment:
Do immutable version of this, as suggested above.

Reply:
Done https://github.com/Joystream/joystream/pull/2512/commits/bc2e269a578d14bdba7cd92bf5b5715eab6ea0d3

Reply:
Great.